libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -406,13 +409,14 @@ void AC_PosControl::input_pos_xyz(const Vector3p& pos, float pos_offset_z, float
                         -constrain_float(accel_max_z_cmss, 0.0f, 750.0f), accel_max_z_cmss,
                         jerk_max_z_cmsss, _dt, false);
 
-    // update the vertical position, velocity and acceleration offsets
+    // update the position, velocity and acceleration offsets
+    update_xy_offsets();
     update_pos_offset_z(pos_offset_z);
```

Comment:
I am thinking we should not pass pos_offset_z into update_pos_offset_z and instead just use _pos_offset_target.z in that function like we do update_pos_offset_xy.

We would then set _pos_offset_target.z at the top of this function.

This would be more consistent between xy and z.

Reply:
ok thanks.  My concern about changing those lines is it would mean I would need to re-test surface tracking and/or terrain following